src/storage/buffer_manager/buffer_manager.cpp

```diff
@@ -458,6 +458,19 @@ uint64_t BufferManager::freeUsedMemory(uint64_t size) {
     return usedMemory.fetch_sub(size);
 }
 
+void BufferManager::resetSpiller(const main::DBConfig& dbConfig) {
+    KU_ASSERT(dbConfig.spillToDiskTmpFile.has_value());
+    if (dbConfig.readOnly) {
+        throw BufferManagerException("Cannot set spill_to_disk_tmp_file for a read only database!");
+    }
+    if (dbConfig.spillToDiskTmpFile->empty()) {
```

Comment:
Can we add a test case that disabling spiller can fail the copy while it would work with the spiller enabled? I'm not sure what's a good configuration for this, but we might need to consider the snap twitter or amazon dataset and tweak the bm size a bit.

Reply:
Okay. And that should also cover the case where I think it's currently just segfaulting since it's not checking if the spiller is enabled or not.